src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -18,6 +18,8 @@ public sealed class CSharpParseOptions : ParseOptions, IEquatable<CSharpParseOpt
         /// </summary>
         public static readonly CSharpParseOptions Default = new CSharpParseOptions();
 
+        private IReadOnlyDictionary<string, string> _features;
```

Comment:
Can we make this readonly?  

If this is only ever assigned `ImmutableDictionary` can we make it that type?